orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/model/PubSubTrigger.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.model;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+
+@JsonTypeName("pubsub")
+public class PubSubTrigger extends Trigger {
+
+  @Nonnull private final Map<String, Object> attributeConstraints;
+  @Nonnull private final Map<String, Object> payloadConstraints;
+  @Nonnull private final Map<String, Object> constraints;
+  @Nonnull private final List<String> expectedArtifactIds;
+  @Nonnull private final String pubsubSystem;
+  @Nonnull private final String subscriptionName;
+
+  @JsonCreator
+  public PubSubTrigger(
+    @Nonnull @JsonProperty("attributeConstraints")
+      Map<String, Object> attributeConstraints,
+    @Nonnull @JsonProperty("payloadConstraints")
+      Map<String, Object> payloadConstraints,
+    @Nonnull @JsonProperty("constraints") Map<String, Object> constraints,
```

Comment:
AFAIK this field doesn't exist for pubsub triggers, this was renamed to `payloadConstraints`.

Reply:
Have I updated this correctly?

Reply:
Yep LGTM.